futures-util/src/stream/buffer_unordered.rs

```diff
@@ -66,6 +79,15 @@ impl<S> BufferUnordered<S>
         self.stream.get_mut()
     }
 
+    /// Acquires a pinned mutable reference to the underlying stream that this
+    /// combinator is pulling from.
+    ///
+    /// Note that care must be taken to avoid tampering with the state of the
+    /// stream which may otherwise confuse this combinator.
+    pub fn get_pin_mut<'a>(self: PinMut<'a, Self>) -> PinMut<'a, S> {
+        unsafe { PinMut::new_unchecked(&mut PinMut::get_mut_unchecked(self).stream) }.get_pin_mut()
```

Comment:
`self.stream().get_pin_mut()`

Reply:
That isn't the same-- the generated `stream()` functions take self by `&mut PInMut` so that they get automatic reborrowing.

Reply:
Interesting. Well I already suspected that something was up when I started seeing it more than once. I've removed the other GitHub comment about this.

Reply:
Nice! That is better, thanks!